internal/taskfile/include.go

```diff
@@ -0,0 +1,156 @@
+package taskfile
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/spf13/afero"
+	yaml "gopkg.in/yaml.v2"
+)
+
+var (
+	// URLReplacer represents a regexp for generating safe filenames form urls
+	URLReplacer = regexp.MustCompile(`[^a-zA-z0-9]`)
+)
+
+type Includes struct {
+	// Path should be a local path or a remote url
+	Path string
+
+	// Cache specifies a duration how long the remote taskfile should be cached
+	Cache *time.Duration `yaml:"cache"`
+
+	// FlushIncludes is used to flush the whole includes section before processing the
+	// taskfile further.
+	FlushIncludes *bool `yaml:"flush_includes"`
+
+	// Dir would be passed from initial taskfile load and represents the current working directory
+	Dir string
+}
+
+// UnmarshalYAML implements yaml.Unmarshaler interface
+func (inc *Includes) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var path string
+	if err := unmarshal(&path); err == nil {
+		inc.Path = path
+		return nil
+	}
+	var includeStruct struct {
+		Path          string
+		Cache         *time.Duration
+		FlushIncludes *bool `yaml:"flush_includes"`
+	}
+	if err := unmarshal(&includeStruct); err != nil {
+		return err
+	}
+	inc.Path = includeStruct.Path
+	inc.Cache = includeStruct.Cache
+	inc.FlushIncludes = includeStruct.FlushIncludes
+	return nil
+}
+
+func (inc *Includes) ApplyDefaults(def *Includes) {
+	if def.FlushIncludes != nil && inc.FlushIncludes == nil {
+		inc.FlushIncludes = def.FlushIncludes
+	}
+	if def.Cache != nil && inc.Cache == nil {
+		inc.Cache = def.Cache
+	}
+}
+
+// IsURL returns true if the given path seems to be a url
+func (inc *Includes) IsURL() bool {
+	if !strings.HasPrefix(inc.Path, "http") {
+		return false
+	}
+	_, err := url.ParseRequestURI(inc.Path)
+	return err == nil
+}
+
+// LoadTaskfile resolves the referenced include and returns a taskfile
+func (inc *Includes) LoadTaskfile() (*Taskfile, error) {
+	var taskfile *Taskfile
+	var err error
+	if inc.IsURL() {
+		taskfile, err = inc.loadTaskfileHTTP()
+	} else {
+		taskfile, err = inc.loadTaskfileLocal()
+	}
+	if err != nil {
+		return nil, err
+	}
+	if inc.FlushIncludes != nil && *inc.FlushIncludes {
+		taskfile.Includes = map[string]*Includes{}
```

Comment:
The `flush_includes` param works same as ignore error from task, it allows to inlcude taskfiles from projects which could contain a includes section as well. The actual behavior is rejecting those, but i wan't to be able to also include them just without raising an execption. Please take a look at the tests, i've added a case which covers them.

Reply:
Got it.

I think we should still get rid of `flush_includes` and return an error in this case for now. This means less confusion / things to explain in the documentation, etc.

We're likely to implement recursive includes at some point soon, which would make `flush_includes` unnecessary. I kept it out at first because I'm doing things in steps. Feel free to give a shot at it (after completing this PR) if you want. 🙂